### PR TITLE
Tweak CurlHandler SSL not supported messages to be more helpful

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @bartonjs, @karelz

### CODE REVIEW

File: src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.Linux.cs [hunk 1]
```diff
@@ -21,6 +21,8 @@ internal partial class CurlHandler : HttpMessageHandler
     {
         private static class SslProvider
         {
+            private const string RequiredOpenSSLBackendName = "OpenSSL/1.0";
```
Comment (bartonjs) [inline src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.Linux.cs]: Interop.Http.OpenSsl10Description is what is used in this class (line 55/57).  Consider setting this const equal to that const, or just using the existing one.

File: src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.OSX.cs [hunk 2]
```diff
@@ -14,6 +14,8 @@ internal partial class CurlHandler : HttpMessageHandler
     {
         private static class SslProvider
         {
+            private const string RequiredOpenSSLBackendName = "SecureTransport";
```
Comment (bartonjs) [inline src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.OSX.cs]: Similarly: Interop.Http.SecureTransportDescription.  Also, this const shouldn't really have OpenSSL in its name :smile:.

File: src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.Linux.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.Linux.cs]: Ah, I'll just use the existing ones.  Thanks.

File: src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.OSX.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.SslProvider.OSX.cs]: :smile: Will fix.

**Pull request merged**